2/DCX Accomodate request/response pattern and general review

rasendubi opened the pull request:
I am also reverting part of 18dbec0cf15001dd408fc6b9f8e676391ec373aa
as it has changed the meaning of problem-solution section.

Note that this is a breaking change to 2/DCX as response format is
changed.

rasendubi committed:
2/DCX Accomodate request/response pattern and general review

I am also reverting part of 18dbec0cf15001dd408fc6b9f8e676391ec373aa
as it has changed the meaning of problem-solution section.

Note that this is a breaking change to 2/DCX as response format is
changed.

AndrewPasika commented on 0002-data-collection-extension/README.md:
```diff
@@ -179,15 +186,20 @@ A processing confirmation response MUST be a UTF-8 encoded JSON record with the
     }
   },
-  "required": [ "id", "status" ],
+  "required": [ "id", "statusCode", "reasonPhrase" ],
```
`reasonPhrase` should be optional as, for example, there is no need to duplicate standard 500 HTTP status code's reason phrase. It only takes additional traffic.

rasendubi committed:
2/DCX Remove unnecessary details for MQTT/CoAP implementation

AndrewPasika commented on 0002-data-collection-extension/README.md:
```diff
@@ -6,88 +6,94 @@ editor: Alexey Shmalko <ashmalko@cybervisiontech.com>
+- The server should know the endpoint that generates the data.
+
+  Solutions:
+  - Using `<endpoint_token>` embedded into resource path.
```
I suggest to change it to:
```
Solutions:
 - Make 2/DCX to be an endpoint-aware extension as defined per 1/KP.
```

rasendubi committed:
2/DCX Address review comments

vtkhir commented on 0002-data-collection-extension/README.md:
```diff
@@ -97,97 +102,21 @@ The payload MUST be a UTF-8 encoded JSON object with the following structure:
-
-JSON Schema for responses is defined in the [response.schema.json](./response.schema.json) file.
+#### Response
+A processing confirmation response MUST follow the error response format as defined per 1/KP.
```
I think it's better to add link here to corresponding section. + I think it makes no sense to return `statusCode` and `reasonPhrase` in case of successful processing

rasendubi replied:
Ok. We can return an empty response then.

rasendubi committed:
2/DCX Make processing confirmation response empty

A success/failure is already carried by the request/response defined
in 1/KP, so adding statusCode and reasonPhrase to success response has
no additinal benefit.

Make processing confirmation response payload empty.
